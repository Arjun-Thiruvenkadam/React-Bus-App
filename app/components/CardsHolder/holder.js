/**
 *
 * Holder
 *
 */

import styled from 'styled-components';

const Holder = styled.div`
  font-family: 'Montserrat', sans-serif;
  font-size: medium;
  margin-top: 10%;
  line-height: 2;
  background-color: white;
`;

export default Holder;
