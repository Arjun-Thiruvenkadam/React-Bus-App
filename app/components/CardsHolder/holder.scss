.CardsHolder {
  font-family: 'Montserrat';
  font-size: large;
  margin-top: 10%;
  line-height: 2;
  background-color: white;
}

.Navlink {
  color: #0f3460;
  font-size: medium;
  background-color: #66bfbf;
  text-decoration: none;
  width: 100%;
  padding: 10px 10px;
  transition-duration: 0.3s;
  transition-property: all;
}

.Navlink:hover {
  font-size: large;
  width: 85px;
  height: 40px;
}
