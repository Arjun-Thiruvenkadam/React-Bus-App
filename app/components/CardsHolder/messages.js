/*
 * CardsHolder Messages
 *
 * This contains all the text for the CardsHolder component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.CardsHolder';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: "Let's start your journey",
  },
  card1: {
    id: `${scope}.card1`,
    defaultMessage:
      'We provide only the best for our customers.We take extra measures to ensure your safe travel.',
  },
  card2: {
    id: `${scope}.card2`,
    defaultMessage:
      'We always give you the lowest price with the best partner offers.',
  },
  book: {
    id: `${scope}.book`,
    defaultMessage: 'Book Tickets',
  },
});
