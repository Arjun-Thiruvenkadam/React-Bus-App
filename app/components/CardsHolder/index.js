/**
 *
 * CardsHolder
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import { NavLink } from 'react-router-dom';
import messages from './messages';
import Card from '../Card/index';
import Safety from '../../images/safety.png';
import Price from '../../images/price.png';
import './holder.scss';

const CardsHolder = () => (
  <div className="CardsHolder">
    <br />
    <h3>
      <FormattedMessage {...messages.header} />
    </h3>
    <br />
    <Card image={Safety} name="Safety">
      <FormattedMessage {...messages.card1} />
    </Card>
    <Card image={Price} name="Price">
      <FormattedMessage {...messages.card2} />
    </Card>
    <br />
    <br />
    <NavLink className="Navlink" to="/order">
      <FormattedMessage {...messages.book} />
    </NavLink>
    <br />
    <br />
  </div>
);

export default CardsHolder;
