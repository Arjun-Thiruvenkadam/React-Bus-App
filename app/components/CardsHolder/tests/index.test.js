import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { MemoryRouter } from 'react-router';
import CardsHolder from '../index';

describe('<CardsHolder />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <MemoryRouter>
        <IntlProvider locale="en">
          <CardsHolder />
        </IntlProvider>
      </MemoryRouter>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
