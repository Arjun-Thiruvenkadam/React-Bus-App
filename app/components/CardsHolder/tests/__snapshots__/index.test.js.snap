// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardsHolder /> should render and match the snapshot 1`] = `
<div
  class="CardsHolder"
>
  <br />
  <h3>
    <span>
      Let's start your journey
    </span>
  </h3>
  <br />
  <div
    class="Card"
  >
    <img
      alt="Safety"
      src="IMAGE_MOCK"
    />
    <h3>
      Safety
    </h3>
    <p>
      <span>
        We provide only the best for our customers. We take extra measures to ensure your safe travel.
      </span>
    </p>
  </div>
  <div
    class="Card"
  >
    <img
      alt="Price"
      src="IMAGE_MOCK"
    />
    <h3>
      Price
    </h3>
    <p>
      <span>
        We always give you the lowest price with the best partner offers.
      </span>
    </p>
  </div>
  <br />
  <br />
  <a
    class="Navlink"
    href="/order"
  >
    <span>
      Book Tickets
    </span>
  </a>
  <br />
  <br />
</div>
`;
